RetailCoder.VBE/UI/Command/Refactorings/RefactorImplementInterfaceCommand.cs

```diff
@@ -29,30 +27,28 @@ public override bool CanExecute(object parameter)
                 return false;
             }
 
-            //var target = _state.FindSelectedDeclaration(Vbe.ActiveCodePane); // nope. logic is a bit more complex here.
+            var selection = Editor.GetSelection();
 
-            var selection = Vbe.ActiveCodePane.GetSelection();
-            var targetInterface = _state.AllUserDeclarations.FindInterface(selection);
+            if (!selection.HasValue)
+            {
+                return false;
+            }
+
+            var targetInterface = _state.AllUserDeclarations.FindInterface(selection.Value);
 
             var targetClass = _state.AllUserDeclarations.SingleOrDefault(d =>
                         !d.IsBuiltIn && d.DeclarationType == DeclarationType.ClassModule &&
-                        d.QualifiedSelection.QualifiedName.Equals(selection.QualifiedName));
-
-            var canExecute = targetInterface != null && targetClass != null;
+                        d.QualifiedSelection.QualifiedName.Equals(selection.Value.QualifiedName));
 
-            Debug.WriteLine("{0}.CanExecute evaluates to {1}", GetType().Name, canExecute);
-            return canExecute;
+            return targetClass != null && targetInterface != null;
         }
 
         public override void Execute(object parameter)
         {
-            if (Vbe.ActiveCodePane == null)
-            {
-                return;
-            }
-
             var refactoring = new ImplementInterfaceRefactoring(_state, Editor, new MessageBox());
-            refactoring.Refactor();
+
+            // ReSharper disable once PossibleInvalidOperationException
+            refactoring.Refactor(Editor.GetSelection().Value);
```

Comment:
The selection could be cached from `CanExecute` instead of re-computed every time.